pandas/core/reshape/pivot.py

```diff
@@ -309,15 +309,17 @@ def _add_margins(
 
     row_names = result.index.names
     # check the result column and leave floats
+
+    def _cast_types(x, dtype):
+        return maybe_downcast_to_dtype(x._values, dtype)
```

Comment:
Should we do that inside `maybe_downcast_to_dtype` instead? That function is also used in some other places (I don't know if it is guaranteed to be called with a non-Series in those other places, but our test coverage might also be lacking on that aspect)

Reply:
sure